src/translate/mod.rs

```diff
@@ -0,0 +1,129 @@
+use lazy_static::lazy_static;
+use regex::Regex;
+use reqwest::header::USER_AGENT;
+use std::str::FromStr;
+use strum::ToString;
+
+lazy_static! {
+    static ref DOUBLE_QUOTES: Regex = Regex::new(r#""(.*?)""#).unwrap();
+}
+
+/// Languages that can used for input and output of the [`translate`] function.
+#[derive(Debug, Clone, PartialEq, Copy, Hash, ToString)]
+pub enum Language {
+    English,
+    Farsi,
+    Arabic,
+    Chinese,
+    French,
+    German,
+    Italian,
+    Japanese,
+    Portuguese,
+    Russian,
+    Spanish,
+}
+
+impl Language {
+    /// Return the language with the language code name. (ex. "ar", "de")
+    pub fn as_code(&self) -> &'static str {
+        match self {
+            Language::English => "en",
+            Language::Farsi => "fa",
+            Language::Arabic => "ar",
+            Language::Chinese => "zh",
+            Language::French => "fr",
+            Language::German => "de",
+            Language::Italian => "it",
+            Language::Japanese => "ja",
+            Language::Portuguese => "pt",
+            Language::Russian => "ru",
+            Language::Spanish => "es",
+        }
+    }
+
+    /// Create a Language from &str like "en" or "French". Case Doesn't matter.
+    pub fn from(s: &str) -> crate::Result<Self> {
+        return Self::from_str(s).map_err(|e| e.into());
```

Comment:
```suggestion
        Self::from_str(s).map_err(|e| e.into())
```

(where is this used? why not use FromStr directly?)